public/lesswrong-editor/src/llm-autocomplete.ts

```diff
@@ -0,0 +1,184 @@
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+// import { ButtonView } from '@ckeditor/ckeditor5-ui';
+import TurndownService from 'turndown';
+
+
+export default class AIAutocomplete extends Plugin {
+    static get pluginName() {
+        return 'AIAutocomplete';
+    }
+
+    init() {
+        const editor = this.editor;
+
+        // editor.ui.componentFactory.add('autocompleteButton', () => {
+        //     const button = new ButtonView();
+
+        //     button.set({
+        //         label: 'AI Autocomplete',
+        //         tooltip: true,
+        //         withText: true
+        //     });
+
+        //     button.on('execute', () => {
+        //         this.autocomplete();
+        //     });
+
+        //     return button;
+        // });
+
+        // Add keyboard shortcut
+        editor.editing.view.document.on('keydown', (evt, data) => {            
+            // Check for Cmd+E (Mac) or Ctrl+E (Windows/Linux)
+            if ((data.ctrlKey || data.metaKey) && data.keyCode === 69) {
+                evt.stop();
+                this.autocomplete();
+            }
+        });
+    }
+
+    autocomplete() {
+        const editor = this.editor;
+        const selection = editor.model.document.selection;
+
+        let selectedContent = this.getSelectedContent();
+
+        // Check the current running document and search for the title of the post using the className .form-component-EditTitle
+        const titleElement = document.querySelector('.form-component-EditTitle');
+        let title = '';
+        if (titleElement) {
+            title = titleElement.textContent;
+            selectedContent = `# ${title}
+by habryka
```

Comment:
What are these lines?

Reply:
These make it so that you send over the title of the post in your prefix. It does sure have "habryka" hardcoded, which does seem nice to vary